system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/TransferPullEndToEndTest.java

```diff
@@ -167,47 +171,59 @@ void httpPull_dataTransfer_withEdrCache() {
             providerDataSource.verify(request("/source").withMethod("GET"));
         }
 
-        @Test
-        void suspendAndResumeByConsumer_httpPull_dataTransfer_withEdrCache() {
+        private static Stream<Participant> participants() {
+            return Stream.of(CONSUMER, PROVIDER);
+        }
+
+        @ParameterizedTest(name = "Suspend and resume httpPull dataTransfer with Edr cache by {0}")
+        @MethodSource("participants")
+        void suspendAndResume_httpPull_dataTransfer_withEdrCache(Participant participant) {
             var assetId = UUID.randomUUID().toString();
             createResourcesOnProvider(assetId, httpSourceDataAddress());
 
-            var transferProcessId = CONSUMER.requestAssetFrom(assetId, PROVIDER)
+            var consumerTransferProcessId = CONSUMER.requestAssetFrom(assetId, PROVIDER)
                     .withTransferType("HttpData-PULL")
                     .execute();
 
-            CONSUMER.awaitTransferToBeInState(transferProcessId, STARTED);
+            CONSUMER.awaitTransferToBeInState(consumerTransferProcessId, STARTED);
 
-            var edr = await().atMost(timeout).until(() -> CONSUMER.getEdr(transferProcessId), Objects::nonNull);
+            var edr = await().atMost(timeout).until(() -> CONSUMER.getEdr(consumerTransferProcessId), Objects::nonNull);
 
             var msg = UUID.randomUUID().toString();
             await().atMost(timeout).untilAsserted(() -> CONSUMER.pullData(edr, Map.of("message", msg), body -> assertThat(body).isEqualTo("data")));
 
-            CONSUMER.suspendTransfer(transferProcessId, "supension");
+            var providerTransferProcessId  = PROVIDER.getTransferProcesses().stream()
+                    .filter(filter -> filter.asJsonObject().getString("correlationId").equals(consumerTransferProcessId))
+                    .map(id -> id.asJsonObject().getString("@id")).findFirst().orElseThrow();
+
+            participant.suspendTransfer(Participant.PROVIDER.equals(participant.getName()) ? providerTransferProcessId : consumerTransferProcessId, "supension");
```

Comment:
this is a demonstration of my theory about using parameterized tests in this case:
conditionals should never be used in tests because it's like adding logic that becomes hard to read and it could lead to further errors.